### PR TITLE
Core/Spells: fix Warrior Revenge proc

### PR DESCRIPTION
**Pull request by HannibalRoG**
**Changes proposed:**

-  Fix proc for warrior revenge it will only become free with dodge or parry.

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)

**Tests performed:** (Does it build, tested in-game, etc.)
Compiled and tested in game

Note:
I did not get it activated against players, maybe it is pvp system problems or issues with parry and dodge stats.

### CODE REVIEW

File: src/server/game/Spells/SpellMgr.cpp [hunk 1]
```diff
@@ -2917,6 +2917,13 @@ void SpellMgr::LoadSpellInfoCorrections()
         spellInfo->AttributesEx3 |= SPELL_ATTR3_CANT_TRIGGER_PROC;
     });
 
+    // Revenge (Warrior) - DBC have SpellLevel & BaseLevel = 0
```
Comment (joschiwald) [inline src/server/game/Spells/SpellMgr.cpp]: what does it fix?

Comment (HannibalRoG) [reply src/server/game/Spells/SpellMgr.cpp]: With only sql for proc with level <22 buff for revenge free appear in list of auras and get you free revenge, revenge is for level 22, that correct baselevel 0 in dbc.

**Comment by HannibalRoG** [comment]
@Shauren @joschiwald @Treeston can you tell me if this is valid?

File: sql/updates/world/master/RoG_2017_09_22_00_world.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+-- Fix for: 	Revenge - (6572, 5302, 5301)
+-- Class:		Warrior
+-- Spec:		Protection
+DELETE FROM `spell_proc` WHERE `SpellId` = 5301;
```
Comment (tkrokli) [inline sql/updates/world/master/RoG_2017_09_22_00_world.sql]: Recommended TC format for this SQL:
```sql
-- Protection Warrior
DELETE FROM `spell_proc` WHERE `SpellId` = 5301; -- Revenge (6572, 5302, 5301)
INSERT INTO `spell_proc` (`SpellId`,`SchoolMask`,`SpellFamilyName`,`SpellFamilyMask0`,`SpellFamilyMask1`,`SpellFamilyMask2`,`SpellFamilyMask3`,`ProcFlags`,`SpellTypeMask`,`SpellPhaseMask`,`HitMask`,`AttributesMask`,`ProcsPerMinute`,`Chance`,`Cooldown`,`Charges`) VALUES
(5301, 0, 0, 0, 0, 0, 0, 0, 0, 0, 48, 0, 0, 0, 0, 0);
```
- remove apostrophes from the numbers (integers)
- place the `VALUES` row on its separate line (readability)
- keep the top comment line short and to the point
- specific details can be put in comment behind the query

**Comment by HannibalRoG** [comment]
What about this guys?? it has been open for more than a month.

**Comment by Treeston** [comment]
There's still an open TODO?

**Comment by HannibalRoG** [comment]
@Treeston any tips about it?

**Comment by joschiwald** [comment]
add a comment why the dbc correction is needed to code too

File: src/server/game/Spells/SpellMgr.cpp (hunk 1 above)
Comment (tkrokli) [reply src/server/game/Spells/SpellMgr.cpp]: Like joschiwald said in https://github.com/TrinityCore/TrinityCore/pull/20394#issuecomment-340783194,
> add a comment why the dbc correction is needed to code too

before we can change the PR status to [Feedback-PatchFix].

**Comment by tkrokli** [comment]
About what HannibalRoG said in the PR TODO text:
> I did not get it activated against players, maybe it is pvp system problems or issues with parry and dodge stats.

I think he tries to say that he believes it is caused by something external (in the PvP system) and that he does not know if it can be fixed by any changes in this PR. (I might be mistaken, though.)

**Comment by HannibalRoG** [comment]
I already answer that when you asked @joschiwald 

> With only sql for proc, with levels <22 aura for revenge free appear in list of auras and get you free revenge, revenge is for level 22, that correct spelllevel and baselevel 0 in dbc for auras 5301 and 5302.

**Comment by tkrokli** [comment]
Hmm. I wonder how much of that description is needed as extra comment in the .cpp code.

**Comment by Wyreth** [comment]
> Spell has wrong level set in DBC

This should be enough, I think?

**Comment by tkrokli** [comment]
BTW, there is something weird about the SQL ... for some reason, travis-ci has determined that this PR is either spam or in violation of their TOS.

**Comment by HannibalRoG** [comment]
@joschiwald 
If you use only the .sql when you parry or dodge the aura for free revenge is activated, it should only be activated when you have learned the revenge in the spell book and not before, the fix corrects that behavior, it is only activated once you have level 22.

@tkrokli dont worry travis is crazy ;)

**Comment by tkrokli** [comment]
OK, I think you don't see it unless you are logged in to Travis (with your GitHub account or any other). The downside to not having travis-ci checking the PR is that we won't see any errors specific to Linux or in the SQL queries (AppVeyor does not check the SQL part). Anyway, here is a screenshot:

<details>

image

**Comment by HannibalRoG** [comment]
@tkrokli Ok, tell me now what travis said.
@joschiwald are you understood now what we did with spell corrections?

**Comment by tkrokli** [comment]
travis-ci says the same things, no change since your last commit:
```
 #20394 f03a4ef
 about 4 hours ago
Clarify comment for spell corrections.
-
Abuse detected
```
travis-ci only lists the last 25 build requests so I can't see any older ones, only the latest 25.

**Comment by Aokromes** [comment]
things of really outdated fork xd

**Comment by tkrokli** [comment]
Aha, I see. Did not think about that as a reason for travis-ci to dislike the code.

**Comment by HannibalRoG** [comment]
And now can anyone take a look here??????????

**Comment by tkrokli** [comment]
I wonder: Would it be OK for you to copy your current PR changes to a new branch from ( `HannibalRoG:Revenge` ) and then open a new PR from it? Maybe that is what is needed to be able to run the travis-ci testing on your code? Just make sure you create the new branch based on current (November) TC source code.

**Comment by joschiwald** [comment]
@HannibalRoG now i got it, referring the tooltip of spell 6572, it would make sense to create a spellscript that prevents the proc if you don't have this spell instead of this dbc correction

**Comment by HannibalRoG** [comment]
@joschiwald I knew that it could be done both ways, but I did not know which of the two would be better, I decided on that one, if you tell me that the other one is better, we change it.

**Comment by joschiwald** [comment]
We need to look at sniffs

**Pull request closed without merging**